xonsh/tools.py

```diff
@@ -49,13 +49,22 @@ def subproc_toks(line, mincol=-1, maxcol=None, lexer=None, returnline=False):
         pos = tok.lexpos
         if tok.type != 'SEMI' and pos >= maxcol:
             break
-        if len(toks) > 0 and toks[-1].type == 'SEMI':
+        if len(toks) == 0 and tok.type in ('WS', 'INDENT'):
+            continue  # handle indentation
+        elif len(toks) > 0 and toks[-1].type == 'SEMI':
```

Comment:
I think this is the line responsible for the `NoneType` error from having `ls;` inside a block in a script.  What is happening was that there is a `DEDENT` token after the semicolon, so the line gets cleared.  Modifying this condition to be as follows seems to fix:
```
        elif len(toks) > 0 and toks[-1].type == 'SEMI' \
                and pos < maxcol and tok.type not in ('NEWLINE', 'DEDENT', 'WS'):
            toks.clear()
```
I think that's all the conditions we want:
* the last thing we saw is was a semicolon
* this token should still be considered (`pos < maxcol`)
* this token is an actual meaningful token (not `DEDENT` or `NEWLINE` or `WS`)